server/lib/scanners/sonarr/index.ts

```diff
@@ -112,13 +116,50 @@ class SonarrScanner
       for (const season of filteredSeasons) {
         const totalAvailableEpisodes = season.statistics?.episodeFileCount ?? 0;
 
+        if (
+          settings.main.removeUnmonitoredFromRequestsEnabled &&
+          season.monitored === false &&
+          totalAvailableEpisodes === 0
+        ) {
+          // Remove unmonitored seasons from Requests
+          const requestRepository = getRepository(MediaRequest);
+          const seasonRequestRepository = getRepository(SeasonRequest);
+
+          const existingRequests = await requestRepository
+            .createQueryBuilder('request')
+            .innerJoinAndSelect('request.media', 'media')
+            .innerJoinAndSelect('request.seasons', 'seasons')
+            .where('media.tmdbId = :tmdbId', { tmdbId: tmdbId })
+            .andWhere('media.mediaType = :mediaType', {
+              mediaType: MediaType.TV,
+            })
+            .andWhere('seasons.seasonNumber = :seasonNumber', {
+              seasonNumber: season.seasonNumber,
+            })
+            .getMany();
+
+          if (existingRequests && existingRequests.length > 0) {
+            existingRequests.forEach((existingRequest) => {
+              existingRequest.seasons.forEach(async (requestedSeason) => {
+                if (requestedSeason.seasonNumber === season.seasonNumber) {
+                  this.log(
+                    `Removing request for Season ${season.seasonNumber} of ${sonarrSeries.title} as it is unmonitored`
+                  );
+                  await seasonRequestRepository.remove(requestedSeason);
+                }
+              });
+            });
+          }
```

Comment:
Here you remove the season request. However, you don't remove the request itself right? You should remove the request if there is no more season request in it.

Reply:
Oh yes you're right, my bad.